src/foam/nanos/crunch/lite/RequirementsRuleAction.js

```diff
@@ -0,0 +1,73 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch.lite',
+  name: 'RequirementsRuleAction',
+
+  documentation: `
+    Rule action that adds payloads to a Capable object.
+    Can only be used on objects implementing Capable.
+  `,
+
+  implements: [
+    'foam.nanos.ruler.RuleAction'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.CapabilityJunctionPayload',
+    'foam.nanos.alarming.Alarm',
+    'foam.nanos.alarming.AlarmReason',
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'requirements'
+    },
+    {
+      class: 'StringArray',
+      name: 'available',
+      documentation: 'Adds a payload but does not add to capabilityIds'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        agency.submit(x, new ContextAgent() {
+          @Override
+          public void execute(X x) {
+            if ( ! (obj instanceof Capable) ) {
+              ((DAO) x.get("alarmDAO")).put(new Alarm(
+                "foam.nanos.lite.RequirementsAction.objectNotCapable",
+                foam.log.LogLevel.ERROR,
+                AlarmReason.CONFIGURATION
+              ));
+              throw new RuntimeException(
+                "Action can only be applied to Capable objects");
+            }
+
+            var capable = (Capable) obj;
+            var capablePayloadDAO = capable.getCapablePayloadDAO(x);
+
+            for ( String id : getRequirements() ) {
+              var payload = new CapabilityJunctionPayload();
+              payload.setCapability(id);
+              capablePayloadDAO.put(payload);
+            }
+
+            for ( String id : getAvailable() ) {
+              var payload = new CapabilityJunctionPayload();
+              payload.setCapability(id);
+              capablePayloadDAO.put(payload);
+            }
+
+            capable.setCapabilityIds(getRequirements());
```

Comment:
doing this will overide anything already set on CapabilityIds... consider putting into loop